planner/core/find_best_task.go

```diff
@@ -169,7 +169,15 @@ func (p *baseLogicalPlan) findBestTask(prop *property.PhysicalProperty) (bestTas
 		// Next, get the bestTask with enforced prop
 		prop.Items = []property.Item{}
 	}
-	physicalPlans := p.self.exhaustPhysicalPlans(prop)
+	physicalPlans, hasUnmatchedHint := p.self.exhaustPhysicalPlans(prop)
+	if hasUnmatchedHint {
+		// If there is a hint in the plan and the hint cannot match the property,
+		// we enforce this property and try to generate the PhysicalPlan again to
+		// make sure the hint can work.
+		prop.Items = []property.Item{}
+		physicalPlans, _ = p.self.exhaustPhysicalPlans(prop)
```

Comment:
If `prop.Enforced` is `true` at line 161, this line may be hit with enforced once through `findBestTask` at line 164. Which is a redundant calculation with line 172

Reply:
I refactor the implementation of `baseLogicalPlan.findBestTask`. PTAL